docs/development/Creating-Upgrade-Script.adoc

```diff
@@ -0,0 +1,95 @@
+= Creating Upgrade Script =
+
+== Overview ==
+
+This page describes the process to create a new upgrade script for the next release.
+
+It assumes that the `major_version`, `minor_version`, and `update_version` macros
+in link:../../pki.spec[pki.spec] have been updated for the next release.
+
+== Upgrade Folder ==
+
+Upgrade scripts are organized by version numbers in link:../../base/server/upgrade[base/server/upgrade] folder.
+
+The new upgrade script needs to be stored in `base/server/upgrade/<version>` folder
+where the `<version>` is the `<major>.<minor>.<update>` version of the next release.
+
+If the folder does not exist, create the folder.
+
+== Index Number ==
+
+The upgrade scripts for each version are sorted by 2-digit index numbers in the filename (i.e. `<index>-<name>.py`).
+
+To determine the index number of the new upgrade script, find the index number of the last upgrade script
+in `base/server/upgrade/<version>` folder, then increment by `1`.
+
+If this is the first upgrade script for the version, use index number `01`.
+
+== Creating Upgrade Script ==
+
+Once the upgrade folder and the index number have been determined,
+create the upgrade script (i.e. `<index>-<name>.py`) in `base/server/upgrade/<version>` folder:
+
+----
+class <name>(pki.server.upgrade.PKIServerUpgradeScriptlet):
+
+    def __init__(self):
+        super(<name>, self).__init__()
+        self.message = ...
+
+    ...
+----
+
+Make sure the class name matches the `<name>` in the filename.
```

Comment:
Can you use `super(__name__, self).__init__()` to automatically get the right name?

Reply:
I actually had to use `super().__init__()` instead. I've updated the doc and the example. Thanks!